.github/workflows/iroha2-dev-pr.yml

```diff
@@ -146,42 +148,65 @@ jobs:
 
   torii-api-and-client-cli-tests:
     runs-on: [self-hosted, Linux, iroha2]
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-04-18
+    #container:
+    # image: hyperledger/iroha2-ci:nightly-2024-04-18
     timeout-minutes: 60
     steps:
       - uses: actions/checkout@v4
       - uses: Swatinem/rust-cache@v2
-      - name: Build binaries
+      - name: Set up Docker Buildx
+        id: buildx
+        if: always()
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+      - name: Install Python and Poetry
         run: |
-          cargo build -p iroha_client_cli -p kagami -p irohad
-      - name: Setup test Iroha 2 environment on the bare metal
+          yum install -y python3.11
+          python3.11 --version
+          curl -sSL https://install.python-poetry.org | python3 -
+          export PATH="/opt/poetry/bin:$PATH"
+      - name: Build and export to Docker iroha:local and iroha:dev images
+        uses: docker/build-push-action@v6
+        if: always()
+        with:
+          context: .
+          load: true
+          file: Dockerfile
+          tags: |
+            hyperledger/iroha:local
+            hyperledger/iroha:dev
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Test docker-compose.yml containers
```

Comment:
```suggestion
      - name: Test docker-compose.single.yml before pushing
        run: |
          docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.single.yml up --wait || exit 1
          docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.single.yml down
      - name: Test docker-compose.local.yml before pushing
        run: |
          docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.local.yml up --wait || exit 1
          docker compose -f ${{ env.DOCKER_COMPOSE_PATH }}/docker-compose.local.yml down
      - name: Test docker-compose.yml containers
```

Reply:
made it in the separate job 
test-docker-compose-before-push: